### PR TITLE
refactor: drop_schedule and models.math with mypy-compliant typing

### PR DESCRIPTION
**Pull request by congbrian**
## What does this PR do?

drop_schedule: Graduates the module off the mypy ignore list and expands the Google-style docstring. Refactors early/late schedule construction (np.zeros + slice assignment instead of np.concatenate) so strict mypy can verify the np.ndarray return type. No intended behavior change.

models.math: Adds MLP annotations and modernizes builtins per PEP 585/604. Replaces @torch.no_grad() with with torch.no_grad(): in accuracy() to satisfy strict mypy across our torch>=2.2.0 range without a `# type: ignore[untyped-decorator]` (fixed upstream in PyTorch 2.10.0, pytorch/pytorch#166413). Add `cast()` on `interpolate()` to satisfy mypy strict `no-any-return`.

**Related Issue(s):** #564 

## Type of Change

- Refactoring (no functional changes)

## Testing

- `uv run --no-sync pytest src/ tests/ -n 1 -m "not gpu" --ignore=tests/run_smoke_all_models.py --timeout=240 --durations=50` — **2635 passed**, 40 skipped (~7m)
- `uv run --no-sync mypy src/rfdetr/ --no-error-summary` — clean (matches `ci-typing.yml`; validates `drop_schedule` and `math` off the mypy ignore list)

- [x] I have tested this change locally
- [ ] I have added/updated tests for this change (existing tests pass)

**Test details:**
- No new tests added; existing `tests/training/callbacks/test_drop_path_callback.py` covers `drop_scheduler` behavior (standard / early / late modes and multi-step rates)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code where necessary, particularly in hard-to-understand areas
- [x] My changes generate no new warnings or errors
- [x] I have updated the documentation accordingly (if applicable)

## Additional Context

`@torch.no_grad()` decorator in math can also be made compliant with strict mypy via `# type: ignore[untyped-decorator]`, however this would then need to be removed if/when torch updated to torch>=2.10.0; context manager is a more robust fix across a broader range of torch versions.

### CODE REVIEW

File: src/rfdetr/models/math.py [hunk 1]
```diff
@@ -21,31 +21,31 @@
...
 
-@torch.no_grad()
-def accuracy(output: torch.Tensor, target: torch.Tensor, topk: Tuple[int, ...] = (1,)) -> List[torch.Tensor]:
+def accuracy(output: torch.Tensor, target: torch.Tensor, topk: tuple[int, ...] = (1,)) -> list[torch.Tensor]:
     """Computes the precision@k for the specified values of k."""
-    if target.numel() == 0:
-        return [torch.zeros([], device=output.device)]
-    maxk = max(topk)
-    batch_size = target.size(0)
+    with torch.no_grad():
```
Comment (Borda) [inline src/rfdetr/models/math.py:26]: Let's add a comment on why this was moved from @torch.no_grad()

File: src/rfdetr/training/drop_schedule.py [hunk 2]
```diff
@@ -22,25 +22,42 @@ def drop_scheduler(
...
+    """
     assert mode in ["standard", "early", "late"]
+    total_iters = epochs * niter_per_ep
     if mode == "standard":
-        return np.full(epochs * niter_per_ep, drop_rate)
+        return np.full(total_iters, drop_rate)
 
     early_iters = cutoff_epoch * niter_per_ep
-    late_iters = (epochs - cutoff_epoch) * niter_per_ep
+    result = np.zeros(total_iters, dtype=np.float64)
```
[the remark this answers is not in the document]
Comment (congbrian) [reply src/rfdetr/training/drop_schedule.py:64]: 👍 added guard for invalid cutoff_epoch values in 65005688fddeb42672f197b80d1629e795dee459

File: src/rfdetr/models/math.py (hunk 1 above)
Comment (congbrian) [reply src/rfdetr/models/math.py:26]: Comment added 👍
65005688fddeb42672f197b80d1629e795dee459

**Review by Borda — DISMISSED** [review]
